### ISSUE TITLE
Apply standardscaler to raster stack

### ISSUE THREAD

**Issue by ggous** [stevenpawley/Pyspatialml#40]
Hello,

I have trained a model based on a csv file.

I have used the StandardScaler from scikit.

`scaler = StandardScaler()`
`x_train = scaler.fit_transform(x_train)`

I am saving the scaler:

`pickle.dump(scaler, open("./scaler", "wb"))`

Now, in order to predict, my test data is a stacked tif image.

I am loading:

`preds = pml.Raster('./masked.tif')`

How can I apply the same standardization on this raster stack?

Thanks!

**Comment by stevenpawley** [comment]
Hello,

Sorry for the delayed response. For cases where you would like to apply a specific function to a Raster object, other than what is supported in the provided methods like `Raster.predict`, the general case is to use `Raster.apply`.

That said, your question relating to applying a transformer directly to the Raster object is interesting and it might be nice to add a `transform` method to the Raster object that supports scikit-learn transformers. I'll try to get back to this over the next week or so, but for now just use `Raster.apply`.

**Comment by stevenpawley** [comment]
I've added a new method, `alter` which is a convenience method to apply a fitted scikit-learn transformer to a Raster object. The name `alter` was used because unfortunately `transform` is alter used extensively though the package for the CRS, and also in rasterio.

**Comment by ggous** [comment]
Very nice!
Thanks!

Στις Σάβ, 30 Ιουλ 2022, 07:50 ο χρήστης Steven Pawley <
***@***.***> έγραψε:
